packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts

```diff
@@ -0,0 +1,214 @@
+  zPluginManifestCliOption,
+  zPluginManifestCliCommand,
+  zPluginPackageTCore,
+  zPluginConfigField,
+  zPluginStorageItemSet,
+  zPluginModule,
+  zPlugin,
+  zModuleMessageOptions,
+  zPluginModuleListItem,
+  zPluginModuleList,
+} from "./Plugin.types";
+
+describe("zPluginModuleIDCombo", () => {
+  test("invalid id", () => {
+    const data = "123:aaa";
+    try {
+      zPluginModuleIDCombo.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe("Invalid Plugin ID");
+    }
+  });
+});
+
+describe("zPluginType", () => {
+  test("invalid option", () => {
+    const data = "a";
+    try {
+      zPluginType.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe(
+        "Invalid enum value. Expected 'action-trigger' | 'action-type' | 'database' | 'decoder' | 'encoder' | 'filesystem' | 'hook' | 'navbar-button' | 'page' | 'service' | 'sidebar-button' | 'sidebar-footer-button'"
+      );
+    }
+  });
+});
+
+describe("zPluginPublisher", () => {
+  test("check optional field", () => {
+    const data = {
+      name: "name",
+    };
+    const parsed = zPluginPublisher.parse(data);
+    expect(parsed.domain).toBeUndefined;
+  });
+});
+
+describe("zPluginInstallOptions", () => {
+  test("accepts empty object", () => {
+    const data = {};
+    const parsed = zPluginInstallOptions.parse(data);
+    expect(parsed.start).toBeUndefined;
+    expect(parsed.log).toBeUndefined;
+    expect(parsed.restoreBackup).toBeUndefined;
+  });
+});
+
+describe("zPluginPermission", () => {
+  test("invalid option", () => {
+    const data = "a";
+    try {
+      zPluginPermission.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe(
+        "Invalid enum value. Expected 'device' | 'analysis' | 'action' | 'device-data' | 'cli'"
+      );
+    }
+  });
+});
+
+describe("zPluginManifestCliOption", () => {
+  test("check optional field", () => {
+    const data = {
+      flags: "flags",
+    };
+    const parsed = zPluginManifestCliOption.parse(data);
+    expect(parsed.description).toBeUndefined;
+  });
+});
+
+describe("zPluginManifestCliCommand", () => {
+  test("error if malformed array", () => {
+    const data = {
+      name: "name",
+      arguments: [{ flags: 0 }],
+      file: "file",
+    };
+    try {
+      zPluginManifestCliCommand.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors).toStrictEqual({ arguments: ["Expected string, received number"] });
+    }
+  });
+});
+
+describe("zPluginPackageTCore", () => {
+  test("error if malformed array", () => {
+    const data = {
+      name: "name",
+      types: ["a"],
+    };
+    try {
+      zPluginPackageTCore.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.types).toStrictEqual([
```

Comment:
Don't add all the values in test, it's going to be difficult to change it later on